Lib/test/test_zipimport.py

```diff
@@ -677,24 +682,24 @@ def testBytesPath(self):
 
         zipimport.zipimporter(filename)
         zipimport.zipimporter(os.fsencode(filename))
-        with self.assertWarns(DeprecationWarning):
+        with self.assertRaises(TypeError):
             zipimport.zipimporter(bytearray(os.fsencode(filename)))
-        with self.assertWarns(DeprecationWarning):
+        with self.assertRaises(TypeError):
             zipimport.zipimporter(memoryview(os.fsencode(filename)))
 
     @support.cpython_only
     def testUninitializedZipimporter(self):
         # The interpreter shouldn't crash in case of calling methods of an
         # uninitialized zipimport.zipimporter object.
         zi = zipimport.zipimporter.__new__(zipimport.zipimporter)
-        self.assertRaises(ValueError, zi.find_module, 'foo')
-        self.assertRaises(ValueError, zi.find_loader, 'foo')
-        self.assertRaises(ValueError, zi.load_module, 'foo')
-        self.assertRaises(ValueError, zi.get_filename, 'foo')
-        self.assertRaises(ValueError, zi.is_package, 'foo')
-        self.assertRaises(ValueError, zi.get_data, 'foo')
-        self.assertRaises(ValueError, zi.get_code, 'foo')
-        self.assertRaises(ValueError, zi.get_source, 'foo')
+        self.assertRaises((ValueError, AttributeError), zi.find_module, 'foo')
```

Comment:
I'd opt for removing anything that's not really needed.

Reply:
I just spoke with @brettcannon - if we want to go in the direction of the Python rewrite (which I think we do), then we should take this opportunity to clean up the Python version by 3.8.  But it's up to you whether you want to land this as a high fidelity translation first, then do a second clean up pass.  If it's easy to do here, it would be nice, but it's your call.  Let's just not forget to do the clean up pass later.

Reply:
I prefer to do a second clean up pass later.

Reply:
Cool